api/v3/Order.php

```diff
@@ -127,6 +127,40 @@ function civicrm_api3_order_create(&$params) {
 }
 
 /**
+ * Cancel a Order.
+ *
+ * @param array $params
+ *   Input parameters.
+ *
+ * @return array
+ */
+function civicrm_api3_order_cancel($params) {
+  $contributionStatuses = CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name');
+  $params['contribution_status_id'] = array_search('Cancelled', $contributionStatuses);
```

Comment:
You just need 
$params['contribution_status_id'] = 'Cancelled';

Since you are calling an api function you can just pass in the name of the value